src/tests/backend/specs/regression-db.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const assert = require('assert').strict;
+const db = require('../../../node/db/DB');
+let AuthorManager;
+let expectedEntries = 0;
+// if this file is run directly, init/shutdown must be called
+let standalone = false;
+
+// wait for database queries to finish
+const waitForDB = async () => {
+  const buffer = db.db.db.buffer;
+  let maxTries = 15; // 50ms*15 === 750 ms
+  return new Promise((resolve, reject) => {
+    const interval = setInterval(async () => {
+      if (buffer._cache.size === expectedEntries) {
+        clearInterval(interval);
+        resolve();
+      } else {
+        if (maxTries === 0) {
+          reject(new Error('not all values written'));
+          clearInterval(interval);
+        }
+        maxTries--;
+      }
+    }, 50);
+  });
+};
+
+describe('regression test for missing await when calling db methods (#5000)', function () {
+  let tmp;
+  before(async function () {
+    // other tests may have called init already
+    if (!db.db) {
+      standalone = true;
+      await db.init();
+    }
+    tmp = db.set;
+    db.set = async (...args) => {
+      // delay db.set
+      await new Promise((resolve) => { expectedEntries++; setTimeout(() => resolve(), 500); });
+      tmp(...args);
+    };
+    // need to be required after changing db.set
+    AuthorManager = require('../../../node/db/AuthorManager');
+  });
+
+  after(async function () {
+    // ensure db is closed after all writes have been written
+    await waitForDB();
+    db.set = tmp;
+    if (standalone) {
+      await db.shutdown();
+    }
+  });
```

Comment:
None of this is necessary if you do `await common.init();`.

Reply:
I do use `common.init()` now and removed everything except `db.set = tmp` to undo the delayed set. Thanks for pointing that out!